### DISCUSSION TITLE
ESP-01 relay board

### DISCUSSION THREAD

**Discussion by mknerr** [RavenSystem/esp-homekit-devices#1907] [MEPLHAA Scripts]
Hello! I'm working on a simple switch for a gas fireplace. Using a esp-01 relay board and a wall toggle switch to trigger the relay.
By adjusting the GPIOs in the Sonoff examples, I have it working, but in the "normally closed" connection to the relay. However, this would mean that if power to the board failed, the fire would be turned on. I would rather it "fail safe".
**How can I adjust this to allow connection to the "normally open" side of the relay?** I believe I need to use "invert" somewhere, but what I've tried hasn't worked.

Here is what I have so far:
`{"c":{"l":1,"b":[{"g":3,"t":5}]},"a":[{"0":{"r":[{"g":0}]},"1":{"r":[{"g":0,"v":1}]},"b":[{"g":3},{"g":2},{"g":2,"t":0}]}]}`

The button coding for GPIO03 can be ignored for now. My intent is to have the physical button on the relay board invoke setup, as well as also operate the relay, but I believe I have to modify the board slightly to make use of the button, since it is designed to reset the board.

**Comment by ceg117** [comment]
Can you provide your route diagram and the modules used?

**Reply by mknerr** [reply]
Thanks for your reply. I apologize for not responding sooner; I did not see a notification for your reply.
I'm not entirely sure what info you require, but I made this graphic to hopefully get started.
Thanks!

![fireplace switch 001](https://user-images.githubusercontent.com/25649210/201180726-256a35df-4184-42ad-af77-f2562b7f2f7d.jpeg)

_Edited to update diagram image_

**Reply by mknerr** [reply]
So, I believe I have solved the main issue above, by swapping the relay values for "off" and "on". Below is what I've come up with at this point. I removed the references to the relay board's built in switch, and also corrected the typo in the LED GPIO, which is GPIO 13. The image posted above has been updated as well.
Currently, the relay board's LED (GPIO13) is lit when the fireplace is on. The ESP01's LED is lit alternately when the toggle switch changes, but does not necessarily reflect the state of the fireplace. I don't believe there is a way to address this LED separately, as it's controlled by GPIO2.

`{
  "c": {
    "l": 13,
    "n": "fireplace-switch" 
  },
  "a": [
    {
      "0": {
        "r": [
          {
            "g": 0,
            "v": 1
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 0,
            "v": 0
          }
        ]
      }, 
      "b": [
        {
          "g":2
        },
        {
          "g":2,
          "t":0
        }
      ]
    }
  ]
}`
